docker-compose.yml

```diff
@@ -83,7 +83,10 @@ services:
     environment:
       - AWS_ACCESS_KEY_ID=${OBJECTSTORE_ACCESSKEY}
       - AWS_SECRET_ACCESS_KEY=${OBJECTSTORE_SECRETKEY}
-    command: s3api create-bucket --bucket cwa --endpoint-url http://objectstore:8000 --acl public-read
+    entrypoint: ["/root/scripts/wait-for-it/wait-for-it.sh", "http://objectstore:8000", "-q","-t", "30", "--"]
```

Comment:
Shouldn't this be "objectstore:8000" instead of "http://objectstore:8080"??
I would also prefer to have it not running silently, thus to remove -q

Reply:
I put it in silent mode, due to the misleading output:

create-bucket_1  | wait-for-it.sh: waiting 30 seconds for objectstore:8000
create-bucket_1  | wait-for-it.sh: timeout occurred after waiting 30 seconds for objectstore:8000

Furthermore, I made the experience in another project that I had to wait for the service at the address to be available, instead of the container and since aws calls http://objectstore:8000 I thought this could be the case here, too. But I tested in with "objectstore:8000" and this works just fine, too.

Reply:
I am trying to find the part with the distribution service that has to be fixed. If I run docker-compose up the distribution service starts and after the bucket was created it just exchanges stuff and exits with code 0. Did this a few times now and always the same. Am I missing something?

Reply:
From my perspective the output wasn't misleading - should be fine after you changed it to objectsore:8080, isn't it?
On some machines the distribution service runs before the bucket was created. That's only the case on the first run. In theory the distribution run should only happen after the bucket was created. But that's more tricky to tackle and I just wanted to understand if you have any idea in that regards.

Reply:
i am trying to reproduce this, but on my machine the distribution service takes a few seconds to the application and the bucket ist created before. I only get a crashing distribution service, if the bucket is missing, hence, if the bucket container crashes before the objectstore is reachable. My first idea would be to ping the create_bucket container until it stopped and start the distribution service afterwards. However, this should be a separate  issue then, since this maybe needs adjustments in the Dockerfile.